plugins/filter/acl_popper.py

```diff
@@ -0,0 +1,339 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2023 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+"""
+The acl_popper filter plugin
+"""
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    name: acl_popper
+    author: Sagar Paul (@KB-perByte)
+    version_added: "2.5.0"
+    short_description: Remove ace entries from a acl source of truth.
+    description:
+        - This plugin removes specific keys from a provided data recursively.
+        - Matching parameter defaults to equals unless C(matching_parameter) is explicitly mentioned.
+        - Using the parameters below- C(data|ansible.utils.acl_popper(target([....])))
+    options:
+      data:
+        description:
+        - This option represents a list of dictionaries or a dictionary with any level of nesting data.
+        - For example C(config_data|ansible.utils.acl_popper(target([....]))), in this case C(config_data) represents this option.
+        type: raw
+        required: True
+      filter_options:
+        description: Specify the target keys to remove in list format.
+        type: dict
+        suboptions:
+          remove:
+            description: Specify aggregate address
+            type: str
+          failed_when:
+            description: Specify aggregate mask
+            type: str
+      match_criteria:
+        description: Specify the matching configuration of target keys and data attributes.
+        type: dict
+        suboptions:
+          afi:
+            description: Specify afi
+            type: str
+          acl_name:
+            description: Specify acl_name
+            type: str
+          source_address:
+            description: Set source_address
+            type: str
+          destination_address:
+            description: Set destination_address
+            type: str
+          sequence:
+            description: sequence
+            type: str
+          protocol:
+            description: Set protocol
+            type: str
+          grant:
+            description: grant
+            type: str
+"""
+
+EXAMPLES = r"""
```

Comment:
Can we add an example of a gathered/set_fact/overridden?  I don't think we need to show the full payload, just the workflow...

Reply:
sure!